releasenotes/notes/vf2layout-preset-passmanager-db46513a24e79aa9.yaml

```diff
@@ -0,0 +1,22 @@
+---
+upgrade:
+  - |
+    The preset passmanagers in :mod:`qiskit.transpiler.preset_passmanagers`
+    for all optimization levels have been changed to run the
+    :class:`~qiskit.transpiler.passes.VF2Layout` by default prior to the layout
+    pass. The :class:`~qiskit.transpiler.passes.VF2Layout` pass will quickly
+    check if a perfect layout can be found and superscedes what was previously
+    done for optimization levels 2 and 3 which were using a combination of
+    :class:`~qiskit.transpiler.passes.TrivialLayout` and
+    :class:`~qiskit.transpiler.passes.CSPLayout` to try and find a perfect
+    layout. However, because :class:`~qiskit.transpiler.passes.VF2Layout` is
+    fast even in the failure mode it has been added to level 1 as
+    well to improve the quality of the results. This will result in potentially
+    different behavior when :func:`~qiskit.compiler.transpile` is called by
+    default as it removes a default path for all optimization levels >=1
+    of using a trivial layout (where ``circuit.qubits[0]`` is mapped to physical
+    qubit 0, ``circuit.qubits[1]`` is mapped to physical qubit 1, etc) assuming
+    the trivial layout is perfect. If your use case was dependent on the
+    trivial layout you can explictly request it when transpiling by specifying
+    ``layout_method="trivial"`` when calling :func:`~qiskit.compiler.transpile`.
+
```

Comment:
I think this might be better expressed as a few separate notes:
1. `CSPLayout` is replaced by `VF2Layout` in the `optimization_level={2,3}` preset pass managers because it's equivalent and significantly faster. (Performance improvement, and no expected behavior change)
2. For `optimization_level=1`, the prior behavior of implicitly trying and using `TrivialLayout` if it pre-routes the circuit has been removed. This behavior continues for `optimization_level=0`. (Behavior change, possibly a deprecation?)
3. For `optimization_level=1`, `VF2Layout` will run prior the default layout selection method and will be used if it finds a layout that pre-routes the circuit. (Behavior change)

2 and 3 could arguably be combined, but I think it's reasonable to separate them since the "try `TrivialLayout` first" behavior, if memory serves, was kept in `level=1` for backwards compatibility. That said, this does seem a good time for it to be removed as it is better covered by either using `optimization_level=0` or `layout_method='trivial'`. Is it reasonable to keep the `TrivialLayout` check (prior to `VF2Layout`) on `optimization_level=1` and raise a `DeprecationWarning` if it ever succeeds, pointing out that the indexes of the found `Layout` may change in future releases?

Reply:
It might make sense, that's what was causing most of the test failures I fixed in: https://github.com/Qiskit/qiskit-terra/pull/7213/commits/4a74c1ff19eba27b9d6fdd54f8186c6895e07ab8 and also it's an indirect cause of the failures we're still hitting in CI (which are caused by vf2 picking the worst qubits on fake backends when there are multiple available mappings and we run with noise and getting a wrong result)

Reply:
I added initial layout back to level 1 and emit a warning when a trivial layout is used in: https://github.com/Qiskit/qiskit-terra/pull/7213/commits/890aeb8c0cb3b42164adebfbe558e5b97b96a978 and updated the release notes in: https://github.com/Qiskit/qiskit-terra/pull/7213/commits/729971f849ff55c872047ccd86d3c87e516c2eb2